draft-ietf-mls-protocol.md

```diff
@@ -1447,6 +1447,42 @@ identifiers in the new credential is valid as a successor to the set of
 presented identifiers in the old credential, according to the application's
 policy.
 
+### Credential Expiry and Revocation
+
+In some credential schemes, the validity status of a credential can change over
+time.  For example, X.509 certificates can expire or be revoked by their issuer.
+These changes can cause operational problems in light of the validation
+requirements of {{credential-validation}}.  Applications can apply some
+operational practices and adaptations to Authentication Service policies to
+moderate these impacts.
+
+It can sometimes be important to distinguish this type of invalidity from other
+types; we call a credential "time-invalid" that is invalid for some time-based
+reason but otherwise valid.  An expired certificate issued by a trusted
+certificate authority would be time-invalid.
+
+In general, to avoid operational problems such as new joiners rejecting expired
+credentials in a group, applications that use such credentials should ensure
+that all of the credentials in use in a group are valid.
+
+If a member finds that its credential is invalid (or will be soon), it should
+issue an Update or Commit that replaces it with a valid credential.  For this
+reason, members SHOULD accept Update proposals and Commits issued by members
+with time-invalid credentials, if the credential in the Update or Commit is
```

Comment:
I still find "time-invalid" to be awkward.  I think "recently expired" is clearer, narrower, and a better fit. There is no reason for example to accept an Update with a credential which is not yet valid (which sounds time-invalid to me).

Reply:
I had shied away from "recently expired" because a credential can also become invalid due to revocation.  I guess we could write "expired or revoked" (since you usually can't tell "recently revoked").

Reply:
I would argue that you do NOT want to accept revoked credentials. That client is dead to me.  (talk to the ✋)